blockchain/validators/src/metadata_key/config.rs

```diff
@@ -0,0 +1,268 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+//! Config for validators.
+
+use super::ParseError;
+use hex::ToHex;
+use mc_blockchain_types::BlockIndex;
+use mc_crypto_keys::Ed25519Public;
+use serde::{Deserialize, Serialize};
+use std::{
+    collections::BTreeMap,
+    ffi::OsStr,
+    fs,
+    ops::RangeInclusive,
+    path::{Path, PathBuf},
+};
+
+/// Container for keys with a range of block indexes that are valid per key.
+pub type KeyValidityMap = BTreeMap<Ed25519Public, Vec<RangeInclusive<BlockIndex>>>;
+
+/// Container for key validity configs.
+/// Supports parsing a `metadata-signers.toml` as specified in [MCIP #43](https://github.com/mobilecoinfoundation/mcips/pull/43).
+#[derive(Clone, Debug, Deserialize, Eq, Ord, PartialEq, PartialOrd)]
+pub struct Config {
+    /// The node key validity ranges.
+    #[serde(alias = "node")] // alias for TOML parsing
+    pub configs: Vec<NodeKeyValidity>,
+
+    /// Optional base path for parsing relative PEM file paths.
+    #[serde(skip)]
+    pub base_path: Option<PathBuf>,
+}
+
+impl Config {
+    /// Load the config TOML from the given file path.
+    pub fn load(path: impl AsRef<Path>) -> Result<Self, ParseError> {
+        let path = path.as_ref();
+        let mut config: Config = match path.extension().and_then(OsStr::to_str) {
+            Some("toml") => Ok(toml::from_slice(&fs::read(path)?)?),
+            _ => Err(ParseError::UnrecognizedExtension(path.into())),
+        }?;
+        config.base_path = path.parent().map(Into::into);
+        Ok(config)
+    }
+
+    /// Get a [KeyValidityMap] from this config.
+    pub fn to_validity_map(&self) -> Result<KeyValidityMap, ParseError> {
+        let mut map = KeyValidityMap::new();
+        for config in &self.configs {
+            let key = parse_pem_or_hex(&config.message_signing_pub_key, self.base_path.clone())?;
```

Comment:
Is there a way go from Ed25519Public to hex yet?

Reply:
Ah yes, context is good. What I meant was do we have a nice existing way to serialize and deserialize to hex for readibility in print & serialization to disk?

Reply:
The `Debug` and `Display` impls do use hex encoding. This logic parses as inline hex, inline PEM, or a PEM file path.